libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Solo.cpp

```diff
@@ -52,15 +56,53 @@ void AP_BattMonitor_SMBus_Solo::timer()
         _state.healthy = true;
     }
 
+    // timeout after 5 seconds
+    if ((tnow - _state.last_time_micros) > AP_BATTMONITOR_SMBUS_TIMEOUT_MICROS) {
+        _state.healthy = false;
+        // do not attempt to ready any more data from battery
+        return;
+    }
+
     // read current
     if (read_block(BATTMONITOR_SMBUS_CURRENT, buff, 4, false) == 4) {
-        _state.current_amps = (float)((int32_t)((uint32_t)buff[3]<<24 | (uint32_t)buff[2]<<16 | (uint32_t)buff[1]<<8 | (uint32_t)buff[0])) / 1000.0f;
+        _state.current_amps = -(float)((int32_t)((uint32_t)buff[3]<<24 | (uint32_t)buff[2]<<16 | (uint32_t)buff[1]<<8 | (uint32_t)buff[0])) / 1000.0f;
```

Comment:
@khancyr, in testing it seems like it's a negative number when the battery is discharging, positive while charging.  In ardupilot we always assume the battery is discharging so the number needs to be positive while discharging.